packages/shared/src/features/onboarding/steps/FunnelRegistration.tsx

```diff
@@ -0,0 +1,156 @@
+import type { ReactElement } from 'react';
+import React, { useRef } from 'react';
+import {
+  Typography,
+  TypographyTag,
+  TypographyType,
+} from '../../../components/typography/Typography';
+import { MobileSocialRegistration } from '../../common/components/MobileSocialRegistration';
+import type { SocialProvider } from '../../../components/auth/common';
+import useRegistration from '../../../hooks/useRegistration';
+import {
+  useEventListener,
+  useToastNotification,
+  useViewSize,
+  ViewSize,
+} from '../../../hooks';
+import {
+  AuthEvent,
+  AuthFlow,
+  getKratosFlow,
+  KRATOS_ERROR,
+} from '../../../lib/kratos';
+import { useAuthContext } from '../../../contexts/AuthContext';
+import { labels } from '../../../lib';
+import { AuthEventNames, isNativeAuthSupported } from '../../../lib/auth';
+import { useLogContext } from '../../../contexts/LogContext';
+import { broadcastChannel } from '../../../lib/constants';
+import Logo, { LogoPosition } from '../../../components/Logo';
+import type { LoggedUser } from '../../../lib/user';
+
+interface FunnelRegistrationProps {
+  onSuccess: (isPlus?: boolean) => void;
+}
+
+const supportedEvents = [AuthEvent.SocialRegistration, AuthEvent.Login];
+
+const useRegistrationListeners = (
+  onSuccess: FunnelRegistrationProps['onSuccess'],
+) => {
+  const { displayToast } = useToastNotification();
+  const { refetchBoot } = useAuthContext();
+  const { logEvent } = useLogContext();
+
+  const onProviderMessage = async (e: MessageEvent) => {
+    const isEventSupported = supportedEvents.includes(e.data?.eventKey);
+
+    if (!isEventSupported) {
+      return undefined;
+    }
+
+    if (e.data?.flow) {
+      const connected = await getKratosFlow(AuthFlow.Registration, e.data.flow);
+
+      logEvent({
+        event_name: AuthEventNames.RegistrationError,
+        extra: JSON.stringify({
+          error: {
+            flowId: connected?.id,
+            messages: connected?.ui?.messages,
+          },
+          origin: 'window registration flow error',
+        }),
+      });
+
+      if (
+        [
+          KRATOS_ERROR.NO_STRATEGY_TO_LOGIN,
+          KRATOS_ERROR.NO_STRATEGY_TO_SIGNUP,
+          KRATOS_ERROR.EXISTING_USER,
+        ].includes(connected?.ui?.messages?.[0]?.id)
+      ) {
+        return displayToast(labels.auth.error.existingEmail);
+      }
+
+      return displayToast(labels.auth.error.generic);
+    }
+
+    const bootResponse = await refetchBoot();
+    const user = bootResponse?.data?.user;
+
+    if (!user || !('providers' in user)) {
+      return displayToast(labels.auth.error.generic);
+    }
+
+    const isPlus = (bootResponse?.data?.user as LoggedUser)?.isPlus;
+
+    return onSuccess(isPlus);
+  };
+
+  useEventListener(broadcastChannel, 'message', onProviderMessage);
+
+  useEventListener(globalThis, 'message', onProviderMessage);
+};
+
+const mobileBg =
```

Comment:
Image should be provided as a prop. It should be configurable parameter for the editor